handlers/pipeline.go

```diff
@@ -138,29 +138,45 @@ func PipelineGet(c echo.Context) error {
 
 // PipelineUpdate updates the given pipeline.
 func PipelineUpdate(c echo.Context) error {
-	p := &gaia.Pipeline{}
-	if err := c.Bind(p); err != nil {
+	p := gaia.Pipeline{}
+	if err := c.Bind(&p); err != nil {
 		return c.String(http.StatusBadRequest, err.Error())
 	}
 
 	// Look up pipeline for the given id
-	var foundPipeline *gaia.Pipeline
+	var foundPipeline gaia.Pipeline
 	for pipeline := range pipeline.GlobalActivePipelines.Iter() {
 		if pipeline.ID == p.ID {
-			foundPipeline = &pipeline
+			foundPipeline = pipeline
 		}
 	}
 
-	if foundPipeline == nil {
+	if foundPipeline.Name == "" {
 		return c.String(http.StatusNotFound, errPipelineNotFound.Error())
 	}
 
+	// We're only handling pipeline name updates for now.
 	if foundPipeline.Name != p.Name {
 		// Pipeline name has been changed.
-		err := pipeline.RenameBinary(*foundPipeline, p.Name)
+
+		// Update exec path
+		p.ExecPath = pipeline.GetExecPath(p)
+
+		// Rename binary
+		err := pipeline.RenameBinary(foundPipeline, p.Name)
 		if err != nil {
 			return c.String(http.StatusInternalServerError, errPipelineRename.Error())
 		}
+
+		// Update pipeline in store
+		err = storeService.PipelinePut(&p)
```

Comment:
I think it's better to update the name in ``foundPipeline`` and store this object instead of ``p`` which we got from the client. For example, a hacker could potentially change something else than the name in the struct and that will be saved too.